src/module/actor/character/document.js

```diff
@@ -148,7 +148,7 @@ class PTUTrainerActor extends PTUActor {
                 + (game.settings.get("ptu", "variant.trainerRevamp") ? (system.level.current >= 25 ? 1 : 0) : 0)
         }
 
-        system.ap.max = 5 + Math.floor(system.level.current / 5);
+        system.ap.max = 5 + Math.floor(system.level.current / (game.settings.get("ptu", "variant.trainerRevamp") ? 2.5 : 5)) + (system.modifiers?.maxAp?.total ? system.modifiers?.maxAp?.total  : 0);
```

Comment:
```suggestion
        system.ap.max = 5 + Math.floor(system.level.current / (game.settings.get("ptu", "variant.trainerRevamp") ? 2.5 : 5)) + (system.modifiers?.maxAp?.total ?? 0);
```
Use [Nullish Coalescing](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing) instead for conciser code